### PR TITLE
Upgrade to rails 6.0.4.4

### PR DESCRIPTION
**Pull request by ekowidianto**
## **Rails 6.0.4 Upgrade**
v6.0.4.4 - https://rubyonrails.org/2021/12/15/Rails-6-0-4-4-and-6-1-4-4-have-been-released
v6.0.4.2 - https://rubyonrails.org/2021/12/14/Rails-6-0-4-2-6-1-4-2-and-7-0-0-rc2-have-been-released
v6.0.4.1 - https://rubyonrails.org/2021/8/19/Rails-6-0-4-1-and-6-1-4-1-have-been-released
v6.0.4.0 - https://rubyonrails.org/2021/6/18/this-week-in-rails-rails-6-0-4-lots-of-active-storage-goodies-and-many-quality-of-life-improvements

Relevant issues: 
1. https://github.com/rails/rails/pull/40056
2. https://github.com/rails/rails/issues/40220 not fixed in v6.0.4.x

### CODE REVIEW

**Comment by ekowidianto** [comment]
KIV
rails issue (https://github.com/rails/rails/issues/40220) not fixed in v6.0.4.x

File: app/models/course/group_user.rb [hunk 1]
```diff
@@ -16,6 +16,12 @@ class Course::GroupUser < ApplicationRecord
   belongs_to :course_user, inverse_of: :group_users
   belongs_to :group, class_name: Course::Group.name, inverse_of: :group_users
 
+  scope :order_by_course_user_name, lambda {
+                                      joins('LEFT OUTER JOIN course_users ON '\
+                                            'course_users.id = course_group_users.course_user_id').
+                                        order('name DESC')
```
Comment (zhuhanming) [inline app/models/course/group_user.rb:22]: Hmm is this the same as `course_users.name ASC`?

Comment (ekowidianto) [reply app/models/course/group_user.rb]: oops should be ASC.. updated thanks

File: app/models/course/group_user.rb [hunk 2]
```diff
@@ -16,6 +16,12 @@ class Course::GroupUser < ApplicationRecord
   belongs_to :course_user, inverse_of: :group_users
   belongs_to :group, class_name: Course::Group.name, inverse_of: :group_users
 
+  scope :order_by_course_user_name, lambda {
+                                      joins('LEFT OUTER JOIN course_users ON '\
+                                            'course_users.id = course_group_users.course_user_id').
+                                        order('name ASC')
```
Comment (zhuhanming) [inline app/models/course/group_user.rb:22]: Hmm do we also need to specify that the `name` column is to be read from the `course_users` table?

Comment (ekowidianto) [reply app/models/course/group_user.rb:22]: Not really.. In fact, there's an error if we use course_users.name:
```
ActiveRecord::StatementInvalid:
       PG::DuplicateAlias: ERROR:  table name "course_users" specified more than once
```

Comment (zhuhanming) [reply app/models/course/group_user.rb:22]: Ooo nice, thanks for the clarification!

**Review by zhuhanming — APPROVED** [review]

**Pull request merged**